examples/flavor/kubernetes/flavor.go

```diff
@@ -0,0 +1,360 @@
+		if _, exists := idsMap[id]; exists {
+			return fmt.Errorf("LogicalID %v specified more than once", id)
+		}
+
+		idsMap[id] = true
+	}
+	for id := range attachments {
+		if _, exists := idsMap[id]; !exists && id != AllInstances {
+			return fmt.Errorf("LogicalID %v used for an attachment but is not in group LogicalIDs", id)
+		}
+	}
+
+	// Only EBS attachments are supported.
+	for _, atts := range attachments {
+		for _, attachment := range atts {
+			if attachment.Type == "" {
+				return fmt.Errorf("no attachment type")
+			}
+		}
+	}
+
+	// Each attachment may only be used once.
+	allAttachmentIDs := map[string]bool{}
+	for _, atts := range attachments {
+		for _, attachment := range atts {
+			if _, exists := allAttachmentIDs[attachment.ID]; exists {
+				return fmt.Errorf("Attachment %v specified more than once", attachment.ID)
+			}
+			allAttachmentIDs[attachment.ID] = true
+		}
+	}
+
+	return nil
+}
+
+type templateContext struct {
+	flavorSpec   Spec
+	instanceSpec instance.Spec
+	allocation   group_types.AllocationMethod
+	index        group_types.Index
+	link         types.Link
+	retries      int
+	poll         time.Duration
+	joinToken    string
+}
+
+func (c *templateContext) Funcs() []template.Function {
+	return []template.Function{
+		{
+			Name: "SPEC",
+			Description: []string{
+				"The flavor spec as found in Properties field of the config JSON",
+			},
+			Func: func() interface{} {
+				return c.flavorSpec
+			},
+		},
+		{
+			Name: "INSTANCE_LOGICAL_ID",
+			Description: []string{
+				"The logical id for the instance being prepared.",
+				"For cattle (instances with no logical id in allocations), this is empty.",
+			},
+			Func: func() string {
+				if c.instanceSpec.LogicalID != nil {
+					return string(*c.instanceSpec.LogicalID)
+				}
+				return ""
+			},
+		},
+		{
+			Name:        "ALLOCATIONS",
+			Description: []string{"The allocations contain fields such as the size of the group or the list of logical ids."},
+			Func: func() interface{} {
+				return c.allocation
+			},
+		},
+		{
+			Name:        "INDEX",
+			Description: []string{"The launch index of this instance. Contains the group ID and the sequence number of instance."},
+			Func: func() interface{} {
+				return c.index
+			},
+		},
+		{
+			Name:        "KUBEADM_JOIN_TOKEN",
+			Description: []string{"Returns the kubeadm JoinToken object"},
+			Func: func() (interface{}, error) {
+				return c.joinToken, nil
+			},
+		},
+		{
+			Name:        "KUBE_JOIN_IP",
+			Description: []string{"Returns the kube advertise IP"},
+			Func: func() (interface{}, error) {
+				return c.flavorSpec.KubeJoinIP, nil
+			},
+		},
+		{
+			Name:        "BIND_PORT",
+			Description: []string{"Returns the kubeadm JoinToken object"},
+			Func: func() (interface{}, error) {
+				return c.flavorSpec.KubeBindPort, nil
+			},
+		},
+		{
+			Name:        "NETWORK_ADDON",
+			Description: []string{"Returns the kube network addon"},
+			Func: func() (interface{}, error) {
```

Comment:
How about making this a function that takes the addon type as a parameter?  Then we can name the function `ADDON` and its signature is `func(addonType string) (interface{}, error)` and we only need to search the list of addons and return the path of the first one with the matching type.